src/hist/intervals.py

```diff
@@ -0,0 +1,107 @@
+from typing import Any, Optional
+
+import numpy as np
+from scipy import stats
+
+__all__ = ("poisson_interval", "clopper_pearson_interval", "ratio_uncertainty")
+
+
+# Code originally contributed to coffea by Nicholas Smith
+# Copyright (c) 2018, Fermilab
+def poisson_interval(
+    values: np.ndarray, variances: np.ndarray, coverage: "Optional[float]" = None
+) -> np.ndarray:
+    r"""
+    The Frequentist coverage interval for Poisson-distributed observations.
+
+    What is calculated is the "Garwood" interval,
+    c.f. https://www.ine.pt/revstat/pdf/rs120203.pdf or
+    http://ms.mcmaster.ca/peter/s743/poissonalpha.html.
+    For weighted data, this approximates the observed count by
+    ``values**2/variances``, which effectively scales the unweighted
+    Poisson interval by the average weight.
+    This may not be the optimal solution: see https://arxiv.org/abs/1309.1287
+    for a proper treatment.
+
+    When a bin is zero, the scale of the nearest nonzero bin is substituted to
+    scale the nominal upper bound.
+    If all bins zero, a warning is generated and interval is set to ``values``.
+
+    Args:
+        values: Sum of weights.
+        variances: Sum of weights squared.
+        coverage: Central coverage interval.
+          Default is one standard deviation, which is roughly ``0.68``.
+
+    Returns:
+        The Poisson central coverage interval.
+    """
+    if coverage is None:
+        coverage = stats.norm.cdf(1) - stats.norm.cdf(-1)
+    scale = np.empty_like(values)
+    scale[values != 0] = variances[values != 0] / values[values != 0]
+    if np.sum(values == 0) > 0:
+        missing = np.where(values == 0)
+        available = np.nonzero(values)
+        if len(available[0]) == 0:
+            raise RuntimeWarning(
```

Comment:
```suggestion
            raise RuntimeError(
```

or a better error. Or use the `warnings.warn` module. You can't raise a Warning, that breaks the control flow, so it's not a "Warning".